msbuild/Xamarin.MacDev.Tasks/Tasks/TextureAtlas.cs

```diff
@@ -28,10 +28,10 @@ protected override string ToolName {
 			get { return "TextureAtlas"; }
 		}
 
-		protected override void AppendCommandLineArguments (IDictionary<string, string> environment, CommandLineBuilder args, ITaskItem input, ITaskItem output)
+		protected override void AppendCommandLineArguments (IDictionary<string, string?> environment, List<string> args, ITaskItem input, ITaskItem output)
 		{
-			args.AppendFileNameIfNotNull (input.GetMetadata ("FullPath"));
-			args.AppendFileNameIfNotNull (Path.GetDirectoryName (output.GetMetadata ("FullPath")));
+			args.Add (input.GetMetadata ("FullPath"));
```

Comment:
Any chance there would be a change in behavior if `input.GetMetadata ("FullPath")` is `null`?

Reply:
The `FullPath` metadata should always be present (it's a built-in metadata: https://learn.microsoft.com/en-us/visualstudio/msbuild/msbuild-well-known-item-metadata?view=vs-2022)

In any case, the existing logic wouldn't work anyway if nothing was passed to the external tool, so this isn't more broken than it was.